### DISCUSSION TITLE
Is there some comparisons between iles and astro

### DISCUSSION THREAD

**Discussion by maoxiaoke** [ElMassimo/iles#12] [General]

**Comment by ElMassimo** [comment]
Yes, you can find a [short comparison in the docs](https://iles-docs.netlify.app/faqs#astro).

Moving this issue to Discussions so that folks can ask additional questions or share their experience.

**Reply by dhruvkb** [reply]
@ElMassimo as the differences between the newest Astro (with Vite) and iles have considerable reduced, are there still plans to actively develop and maintain iles?

Not to be rude but wouldn't merging forces with Astro produce a single, far superior solution? There are a lot of cool features in iles that are lacking in Astro and Astro has a huge userbase (I'm basing this on GitHub stars and non-Vue framework support) that could contribute back to the project.

**Reply by ElMassimo** [reply]
Yes, I still plan to work on it since I _very much_ enjoy using it 😃 

---

> wouldn't merging forces with Astro produce a single, far superior solution?

There are trade-offs in the architecture of each project, so I'm afraid it's not that simple.

_îles_ made a bet on Vue single-file-components, Astro made a bet on a _custom_ JSX-based syntax.

_îles_ optimizes for SSG, Astro intends to grow to cover as many use cases as possible (such as SSR).

> There are a lot of cool features in iles that are lacking in Astro

This is possible __because__ îles is built using Vue. If it works with Vue SFC, it probably works with îles.

For example, being able to use almost every Vite plugin that works with Vue, such as [`unplugin-vue-components`](https://github.com/antfu/unplugin-vue-components) and [`UnoCSS`](https://github.com/unocss/unocss).

> Astro has a huge userbase

Vue has a far bigger userbase, and îles is built on top of Vue, so it benefits from a larger ecosystem.

For example, auto-importing components in Astro would require a custom Vite plugin, while îles can use `unplugin-vue-components`.

> non-Vue framework support

îles [supports non-Vue frameworks](https://iles.pages.dev/guide/frameworks).

**Reply by dhruvkb** [reply]
I'm very glad to hear about continued development. The use of 100% Vue is what made me switch from Astro to îles in the first place and the only reason I asked was to ensure that the choice would hold for the foreseeable future.

**Comment by coder0107git** [comment]
How does the comparison still look? The comparison in the docs are fairly outdated by this point.

**Reply by ElMassimo** [reply]
At this point Astro has grown into a medium-sized company with a very large community, to the point where it doesn't even make sense to compare these anymore.

Astro as a framework is significantly more polished than when I wrote this library.

I'd say, if you weren't using îles, go with Astro. If you are already using îles, you can continue using it, the gist of it will remain the same.

**Reply by TechAkayy** [reply]
Thanks @ElMassimo for you suggestion.

While there are architectural differences between Astro & Ies, the most significant one is that Astro uses a jsx-mixed-html DSL while Iles uses Vue SFC to author the layouts & pages. 

As both frameworks uses Vite underneath, having a "pre" Vite Plugin in the Astro vite plugin chain that converts Vue SFC to .astro would make authoring of Astro pages with Vue SFC possible, and enjoy all benefits of Astro.

I understand this might not be simple, but just curious to understand if you ever had this thought?

Personally for me Vue SFC for life, I can't imagine I will ever like JSX hybrid syntaxes (.astro, .svelte etc) , lol!

**Reply by ElMassimo** [reply]
> having a "pre" Vite Plugin in the Astro vite plugin chain that converts Vue SFC to .astro

I'm not familiar enough with Astro's internals, but that sounds interesting.
